MarsAdvancedTaskPart1NUnitAutomationV4/Steps/LanguageSteps.cs

```diff
@@ -0,0 +1,197 @@
+{
+
+    public class LanguageSteps
+    {
+        string actualMessage;
+        string expectedMessage;
+        string languageName;
+        LanguageAddAndUpdateComponent languageAddAndUpdateComponent;
+        LanguageRenderingComponent languageRenderingComponent;
+        LanguageAssertHelper languageAssertHelper;
+        int cancelFlag;
+
+        public LanguageSteps()
+        {
+            actualMessage = string.Empty;
+            expectedMessage = string.Empty;
+            languageName = string.Empty;
+            cancelFlag = 0;
+            languageAddAndUpdateComponent = new LanguageAddAndUpdateComponent();
+            languageAssertHelper = new LanguageAssertHelper();
+            languageRenderingComponent = new LanguageRenderingComponent();
+
+        }
+        public void DeleteLanguageRecords()
+        {
+            languageAddAndUpdateComponent.DeleteAllLanguageRecords();
+            int rowCount = languageRenderingComponent.GetLanguageRecordsCount();
+            languageAssertHelper.AssertDeleteAllLanguageRecords(rowCount);
+        }
+        public void AddNewLanguage(LanguageDM languageDM)
+        {
+            languageAddAndUpdateComponent.DeleteAllLanguageRecords();
+            languageAddAndUpdateComponent.AddLanguage(languageDM);
+            actualMessage = languageRenderingComponent.CapturePopupMessage();
+            languageName = languageRenderingComponent.GetLastLanguageName();
+
+            if (!string.IsNullOrEmpty(languageName))
+                expectedMessage = languageName + " has been added to your languages";
+            else
+                expectedMessage = "has been added to your languages";
+            languageAssertHelper.AssertAddNewLanguage(expectedMessage, actualMessage);
+
+        }
+
+        public void AddNewLanguageRecordWithInsufficientData(LanguageDM languageDM)
+        {
+            languageAddAndUpdateComponent.DeleteAllLanguageRecords();
+            languageAddAndUpdateComponent.CheckAndMakeSpaceAvailablity();
+            languageAddAndUpdateComponent.AddLanguage(languageDM);
+            actualMessage = languageRenderingComponent.CapturePopupMessage();
+            expectedMessage = "Please enter language and level";
+            languageAddAndUpdateComponent.DeleteAllLanguageRecords();
+            languageAssertHelper.AssertNewLanguageNotAddedSuccessfully(expectedMessage, actualMessage);
+        }
+
+        public void AddAlreadyExistingLanguageRecord(LanguageDM languageDM)
+        {
+            languageAddAndUpdateComponent.DeleteAllLanguageRecords();
+            languageAddAndUpdateComponent.AddLanguage(languageDM);
+            languageAddAndUpdateComponent.AddLanguage(languageDM);
+            actualMessage = languageRenderingComponent.CapturePopupMessage();
+            expectedMessage = "This language is already exist in your language list.";
+            languageAddAndUpdateComponent.DeleteAllLanguageRecords();
```

Comment:
so here you should not be deleting all records after you should only be deleting the ones the test has added 


so think about first clean up before any test runs (you delete all data around this feature e.g  before any language test run you delete all languages). Then after each test delete only data that the test has added! (test1 added english and test2 to added french then test1 should only delete english and test2 should delete french only)